### PR TITLE
1615 payment part 1

### PR DESCRIPTION
**Pull request by xsalefter**
Some notes:

1. I accidentally add sonatype snapshot repository when update parent version days ago. This PR remove it.
2. Use `MultiValueMap` instead of `MultiMap` in `PaymentRefresher`. Added test for it in `TestPaymentRefresher`.

### CODE REVIEW

File: payment/src/test/java/org/killbill/billing/payment/core/TestPaymentRefresher.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        Mockito.doReturn(ANY_PAYMENT)
+               .when(toMock).toPayment(Mockito.any(UUID.class),
+                                       Mockito.anyIterable(),
+                                       Mockito.anyBoolean(),
+                                       Mockito.anyBoolean(),
+                                       Mockito.any());
+        return toMock;
+    }
+
+    @Test
```
Comment (sbrossie) [inline payment/src/test/java/org/killbill/billing/payment/core/TestPaymentRefresher.java:114]: Missing the `fast` groups

**Review by sbrossie — APPROVED** [review]

File: payment/src/main/java/org/killbill/billing/payment/api/DefaultPayment.java [hunk 2]
```diff
@@ -93,29 +92,24 @@ public DefaultPayment(final UUID id, @Nullable final DateTime createdDate, @Null
...
-                                                                                                                                    transaction.getTransactionType() == TransactionType.PURCHASE ||
-                                                                                                                                    transaction.getTransactionType() == TransactionType.CREDIT;
-                                                                                                                         }
-                                                                                                                     }), null);
+            transactionToUseForCurrency = transactions
+                    .stream()
+                    .filter(transaction -> transaction.getTransactionType() == TransactionType.AUTHORIZE ||
+                                           transaction.getTransactionType() == TransactionType.PURCHASE ||
+                                           transaction.getTransactionType() == TransactionType.CREDIT)
+                    .findFirst().orElse(null);
```
Comment (pierre) [inline payment/src/main/java/org/killbill/billing/payment/api/DefaultPayment.java:110]: Shouldn't this be something like `findLast()` instead (`getLast` was used before)?

File: payment/src/main/java/org/killbill/billing/payment/core/PaymentRefresher.java [hunk 3]
```diff
@@ -311,17 +304,17 @@ public Payment apply(final PaymentModelDao paymentModelDao) {
                                                if (!withPluginInfo) {
                                                    pluginApi = null;
                                                } else {
-                                                   if (paymentMethodIdToPaymentPluginApi.get(paymentModelDao.getPaymentMethodId()) == null) {
```
Comment (pierre) [inline payment/src/main/java/org/killbill/billing/payment/core/PaymentRefresher.java:314]: Isn't the `null` check necessary to go through the `if` the first time?

**Review by pierre — CHANGES_REQUESTED** [review]

File: payment/src/main/java/org/killbill/billing/payment/core/PaymentRefresher.java (hunk 3 above)
Comment (xsalefter) [reply payment/src/main/java/org/killbill/billing/payment/core/PaymentRefresher.java:314]: Yeah you're right. Sorry.

**Review by pierre — APPROVED** [review]

**Comment by pierre** [comment]
@sbrossie FYI I'll work on updating our old [CyberSource and Janitor](https://github.com/killbill/killbill-integration-tests/tree/67ace7a25962842a532d4c943e5bedc836502096/killbill-integration-tests/plugins/payment/errors-handling) integration tests, to help increase confidence no regression was introduced after these changes.

**Pull request merged**